api-catalog-services/src/main/java/org/zowe/apiml/apicatalog/swagger/ApiDocRetrievalServiceRest.java

```diff
@@ -19,64 +20,54 @@
 import org.apache.hc.core5.http.HttpStatus;
 import org.apache.hc.core5.http.io.entity.EntityUtils;
 import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.http.MediaType;
 import org.springframework.stereotype.Service;
 import org.springframework.web.util.UriComponents;
 import org.springframework.web.util.UriComponentsBuilder;
-import org.zowe.apiml.apicatalog.instance.InstanceRetrievalService;
-import org.zowe.apiml.apicatalog.services.cached.model.ApiDocInfo;
-import org.zowe.apiml.apicatalog.services.status.model.ApiDocNotFoundException;
-import org.zowe.apiml.apicatalog.services.status.model.ApiVersionNotFoundException;
-import org.zowe.apiml.apicatalog.swagger.SubstituteSwaggerGenerator;
+import org.zowe.apiml.apicatalog.exceptions.ApiDocNotFoundException;
+import org.zowe.apiml.apicatalog.exceptions.ApiVersionNotFoundException;
 import org.zowe.apiml.config.ApiInfo;
 import org.zowe.apiml.eurekaservice.client.util.EurekaMetadataParser;
 import org.zowe.apiml.message.log.ApimlLogger;
 import org.zowe.apiml.product.gateway.GatewayClient;
-import org.zowe.apiml.product.instance.InstanceInitializationException;
 import org.zowe.apiml.product.instance.ServiceAddress;
 import org.zowe.apiml.product.logging.annotations.InjectApimlLogger;
 import org.zowe.apiml.product.routing.RoutedService;
 import org.zowe.apiml.product.routing.RoutedServices;
+import org.zowe.apiml.util.EurekaUtils;
 
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.function.UnaryOperator;
 
 /**
  * Retrieves the API documentation for a registered service
  */
 @Service
-@ConditionalOnProperty(
-        value = "apiml.catalog.standalone.enabled",
-        havingValue = "false",
-        matchIfMissing = true)
 @RequiredArgsConstructor
 @Slf4j
-public class APIDocRetrievalService {
+public class ApiDocRetrievalServiceRest implements ApiDocRetrievalService {
+
+    private static final UnaryOperator<String> exceptionMessage = serviceId -> "No API Documentation was retrieved for the service " + serviceId + ".";
 
     @Qualifier("secureHttpClientWithoutKeystore")
     private final CloseableHttpClient secureHttpClientWithoutKeystore;
 
-    private final InstanceRetrievalService instanceRetrievalService;
+    private final EurekaClient eurekaClient;
```

Comment:
Why not refactor this to use the DiscoveryClient interface from spring cloud?

Reply:
this comment applies to all uses of EurekaClient

Reply:
My mistake, you are right, there has to be another refactoring. Are you ok to do it in the next PR? For sure I have added the task to the list. We should do it before including API Catalog as part of apiml module.